### ISSUE TITLE
Update mongoDb instrumentation

### ISSUE THREAD

**Issue by SimunKaracic** [kamon-io/Kamon#994]
The current implementation is using old versions of drivers (a couple of major versions behind).

**Comment by jorkzijlstra** [comment]
When using the Kamon bundle (2.1.8) and the latest mongo-scala-driver (4.2.3) all driver metrics are lost and not being exported anymore by the Prometheus reporter.

It seem that the mongo-scala-driver has changed a lot and uses different Operation classes which probably is the reason why [spanBuilder](https://github.com/kamon-io/Kamon/blob/master/instrumentation/kamon-mongo/src/main/scala/kamon/instrumentation/mongo/MongoClientInstrumentation.scala#L139) is not being hit anymore. I verified this with setting some breakpoints.

When reverting the mongo-scala-driver back to version 2.8.0 everything starts working again the metrics are exposed on the endpoint:
```
span_processing_time_seconds_bucket{db_instance="test",operation="test.find",span_kind="client",parentOperation="/test",environment="local",error="false",le="0.005",component="mongo.driver.sync",mongo_collection="test"} 0.0
span_processing_time_seconds_bucket{db_instance="test",operation="test.find",span_kind="client",parentOperation="/test",environment="local",error="false",le="0.01",component="mongo.driver.sync",mongo_collection="test"} 0.0
span_processing_time_seconds_bucket{db_instance="test",operation="test.find",span_kind="client",parentOperation="/test",environment="local",error="false",le="0.025",component="mongo.driver.sync",mongo_collection="test"} 0.0
sp
... and the rest of the bucket data
```

Gitter details: https://gitter.im/kamon-io/Kamon?at=6075d7ba97cf5067465dbb8e

**Comment by jtjeferreira** [comment]
Maybe an easier implementation of the mongo instrumentation is to use the a `CommandListener`. This is what we are using:

```scala
/**
 * Inspired in https://github.com/opentracing-contrib/java-mongo-driver/blob/master/src/main/java/io/opentracing/contrib/mongo/TracingCommandListener.java
 */
class KamonCommandListener extends CommandListener {

  def collectionName(event: CommandStartedEvent): String = {
    Document(event.getCommand).get[BsonString](event.getCommandName).map(_.getValue).getOrElse("")
  }
  def operationName(event: CommandStartedEvent): String = s"${event.getCommandName}_${collectionName(event)}"

  /**
   * Cache for (request id, span) pairs
   */
  private val cache = new ConcurrentHashMap[Int, Span]()

  override def commandStarted(event: CommandStartedEvent): Unit = {
    val currentContext = Kamon.currentContext()
    val parentSpan = currentContext.get(Span.Key)

    if (!parentSpan.isEmpty) {

      val spanBuilder = Kamon
        .spanBuilder(operationName(event))
        .asChildOf(parentSpan)
        .tagMetrics("span.kind", "client")
        .tagMetrics("component", "scala-mongo")
        .tagMetrics("db.instance", event.getDatabaseName)
      //.withTag("db.statement", event.getCommand.toString)

      val span = spanBuilder.start()

      cache.put(event.getRequestId, span)
    }
    ()
  }

  override def commandSucceeded(event: CommandSucceededEvent): Unit = {
    val span = cache.remove(event.getRequestId)
    if (span != null)
      span.finish()
  }

  override def commandFailed(event: CommandFailedEvent): Unit = {
    val span = cache.remove(event.getRequestId)
    if (span != null)
      span.fail("error.object", event.getThrowable).finish()
  }
}

import kamon.Metrics
class KamonConnectionPoolListener(kamon: Metrics) extends ConnectionPoolListener {

  private val connectionsGauge = kamon.gauge("mongo.connections").withoutTags()
  private val connectionCheckedIns = kamon.rangeSampler("mongo.connectionCheckedIns").withoutTags()

  override def connectionCheckedOut(event: ConnectionCheckedOutEvent): Unit = { connectionCheckedIns.increment(); () }
  override def connectionCheckedIn(event: ConnectionCheckedInEvent): Unit = { connectionCheckedIns.decrement(); () }

  override def connectionCreated(event: ConnectionCreatedEvent): Unit = { connectionsGauge.increment(); () }
  override def connectionClosed(event: ConnectionClosedEvent): Unit = { connectionsGauge.decrement(); () }
}
```

**Comment by SimunKaracic** [comment]
Should be fixed with v2.1.18.

**Comment by jorkzijlstra** [comment]
Thanks, seem to be its indeed working.
